src/Graphing.hs

```diff
@@ -92,13 +92,17 @@ edge parent child gr = gr { graphingAdjacent = adjacent' }
 -- - A way to convert a dependency @toDependency@
 --
 -- __Unfold does not work for recursive inputs__
-unfold :: Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
-unfold seed getDeps toDependency = foldr addNode empty seed
+unfold :: forall dep res. Ord res => [dep] -> (dep -> [dep]) -> (dep -> res) -> Graphing res
+unfold seed getDeps toDependency = Graphing
+  { graphingDirect = S.fromList (map toDependency seed)
+  , graphingAdjacent = AM.vertices (map toDependency seed) `AM.overlay` AM.edges [(toDependency parentDep, toDependency childDep) | (parentDep,childDep) <- edgesFrom seed]
```

Comment:
[nit] Style-wise, when do you suggest using infix notation?  I find it hard to read sometimes, probably due to my more limited experience.

Reply:
depends on the situation -- I usually use it in one of these cases:
1. I'm using a binary function in a section, e.g.,
```hs
(`M.member` someMap)
```
2. A function is named in a way that's meant to be used infix, e.g., `Data.Function.on` or `Text.URI.relativeTo`
3. To combine two larger expressions _of the same type_. `AM.overlay (somereallylongexpression) (anotherreallylongexpression)` is fine too. In general, the intuition around infix functions is that they have very low precedence, so
```hs
foo bar `and` baz

-- is almost always going to be interpreted as
(foo bar) `and` (baz)
```